Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+    def save(self, filename):
+        """Save a Table to a file.
+
+        Parameters
+        ----------
+        filename : str
+            The destination file path, absolute or relative.
+        """
+        ext = os.path.splitext(filename)[1]
+        from Orange.data.io import FileFormat
+        writer = FileFormat.writers.get(ext)
+        if not writer:
+            desc = FileFormat.names.get(ext)
+            if desc:
+                raise IOError("Writing of {}s is not supported".format(desc.lower()))
+            else:
+                raise IOError("Unknown file name extension.")
+        writer.write_file(filename, self)
+
+    def __getitem__(self, item):
+        # if selecting a column subset, we need to transfer weights so they don't just disappear
+        # only do this for multiple column selection, which returns a DataFrame by contract
+        if isinstance(item, (Sequence, pd.Index)) and not isinstance(item, str) \
+                and all(isinstance(i, str) for i in item) \
+                and self._WEIGHTS_COLUMN not in item:
+            item = list(item) + self._INTERNAL_COLUMN_NAMES
+        return super().__getitem__(item)
+
+    def __setitem__(self, key, value):
+        # if the table has an empty index and we're inserting a new row,
+        # the index would be created by pandas automatically.
+        # we want to maintain unique indices, so we override the index manually.
+        # we also need to set default weights, lest they be NA
+        new_index_and_weights = len(self.index) == 0
+
+        # PANDAS CONTRACT BREAKAGE:
+        # the pandas default behaviour when adding a new column as a Series is that
+        # the indexes are inner-joined: only the elements at the indices that exist
+        # in the current table are actually set, other elements are NA
+        # for easier handling of Orange behaviour, we effectively ignore the index
+        # on the series to just merge the column into the table
+        if isinstance(value, pd.Series) and not new_index_and_weights:
+            value.index = self.index
+        super().__setitem__(key, value)
+
+        if new_index_and_weights:
+            new_id = self._new_id(len(self), force_list=True)
+            self.index = new_id
+            # super call because we'd otherwise recurse back into this
+            super().__setitem__(self._WEIGHTS_COLUMN, 1)
+
+    def clear(self):
+        """Remove all rows from the table in-place."""
+        self.drop(self.index, inplace=True)
+
+    def append(self, other, ignore_index=False, verify_integrity=False):
+        """Append a new row to a table, returning a new Table.
+
+        Parameters
+        ----------
+        other : list[obj] | SeriesBase | TableBase.
+            A list-like of a single row, TableSeries (a single row slice) or a Table.
+        ignore_index : bool, optional, default False
+            Provided for pandas API compatibility, ignored.
```

Comment:
Things break in weird ways. I actually checked where this is used and replaced all non-test usages with `Table.concatenate`, then removed this override. Hooray!